src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncExceptionHandlerRewriter.cs

```diff
@@ -516,7 +516,7 @@ private BoundNode VisitCatchBlock(BoundCatchBlock node, AwaitCatchFrame parentAw
             if (!_analysis.CatchContainsAwait(node))
             {
                 var origCurrentAwaitCatchFrame = _currentAwaitCatchFrame;
-                _currentAwaitCatchFrame = null;
+                _currentAwaitCatchFrame = parentAwaitCatchFrame;
```

Comment:
>_currentAwaitCatchFrame = parentAwaitCatchFrame; [](http://example.com/codeflow?start=16&length=48)

Is it necessary to adjust not null ```_currentAwaitCatchFrame``` here? Should we simply leave it as is in that case?

Reply:
Yes, `_currentAwaitCatchFrame` can be the frame of the *previous* `catch` (despite the field's name). When going over `node.CatchBlocks` in `RewriteFinalizedRegion`, `catch`es with `await` will set `_currentAwaitCatchFrame` but never reset it. Otherwise it should be only `null` - again see `RewriteFinalizedRegion` - it's set to `null` there before calling the first `VisitCatchBlock`.